presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeExecution.java

```diff
@@ -70,4 +70,14 @@ public void testFailures()
                 ".*Scalar function name not registered: presto.default.sequence.*");
         assertQueryFails("SELECT orderkey / 0 FROM orders", ".*division by zero.*");
     }
+
+    /**
+     * Test native execution of a cpp function declared via json file, with sample function eq() defined
+     * in src/test/resources/eq.json
+     */
+    @Test
+    public void testJsonFileBasedFunction()
+    {
+        assertQuery(getSession(), "SELECT json.test_schema.eq(1, linenumber) FROM lineitem", "SELECT 1 = linenumber FROM lineitem");
```

Comment:
Is it necessary to pass getSession()? Why can't we just call assertQuery(sql, otherSql) ?

FYI, there is a "clamp" function that's available in Velox, but not in Presto, which we could use instead of "eq". That said, I don't think we have a similar aggregate function.